blockchain/upgrade.go

```diff
@@ -3189,7 +3189,7 @@ func upgradeToVersion10(ctx context.Context, db database.DB, dbInfo *databaseInf
 
 // separateUtxoDatabase moves the UTXO set and state from the block database to
 // the UTXO database.
-func separateUtxoDatabase(ctx context.Context, b *BlockChain) error {
+func separateUtxoDatabase(ctx context.Context, b *BlockChain, utxoDb database.DB) error {
```

Comment:
It looks like the only thing used from the `BlockChain` now is the database.  I think it would probably be nicer to just pass that instead here to limit the scope to exactly what the function needs.  e.g. `separateUtxoDatabase(ctx context.Context, db database.DB, utxoDb database.DB)`.

Reply:
I agree, updated.